src/checkers/htmlchecker.py

```diff
@@ -109,6 +109,24 @@ def check(self, external_data):
 
         self._update_version(external_data, latest_version, abs_url)
 
+    @staticmethod
+    def _substitute_placeholders(template_string, version):
+        version_list = LooseVersion(version).version
+        number_of_fields = len(version_list)
+        template_obj = Template(template_string)
+
+        replacements = {"version": version, "major": version_list[0]}
+
+        for i in range(number_of_fields):
+            replacements["version" + str(i)] = str(version_list[i])
```

Comment:
Use f-strings instead of concatenation, i.e. `f"version{i}"`

Reply:
Yeah, I just saw your suggestion.